test/jdk/com/sun/jdi/TestScaffold.java

```diff
@@ -463,43 +463,65 @@ protected void failure(String str) {
 
     private ArgInfo parseArgs(String args[]) {
         ArgInfo argInfo = new ArgInfo();
-        String mainWrapper = System.getProperty("main.wrapper");
-        if ("Virtual".equals(mainWrapper)) {
-            argInfo.targetAppCommandLine = TestScaffold.class.getName() + " " + mainWrapper + " ";
-            argInfo.targetVMArgs += "--enable-preview ";
-        } else if ("true".equals(System.getProperty("test.enable.preview"))) {
-            // the test specified @enablePreview.
-            argInfo.targetVMArgs += "--enable-preview ";
-        }
-
+        // Parse arguments, like java/j* tools command-line arguments
+        // the first argument not-starting with '-' is treated as a classname
+        // the other arguments are split to targetVMArgs targetAppCommandLine correspondingly
+        // The example of args for line '@run driver Frames2Test -Xss4M' is  '-Xss4M' 'Frames2Targ'.
```

Comment:
Is `Frames2Test` a typo there?

Reply:
Frames2Test is the name of test that uses additional command-line options. It is an example.

Reply:
In that case did you mean:
```
'@run driver Frames2Test -Xss4M Frames2Targ'
```
?

Reply:
the @run line contains only the test name and additional command-line options if needed, the target app class 'Frames2Targ' is not included. I have copy-pasted this example from 
https://github.com/openjdk/jdk/blob/0deb648985b018653ccdaf193dc13b3cf21c088a/test/jdk/com/sun/jdi/Frames2Test.java#L34

Reply:
That is very confusing because the `Frames2Targ` seems to appear out of nowhere.

Reply:
Well, I could update the comment, but it is basically how the all jdi tests work. The test adds target by itself after all command line arguments.